src/webui/www/private/scripts/piecesbar.js

```diff
@@ -0,0 +1,267 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+'use strict';
+
+if (window.qBittorrent === undefined) {
+    window.qBittorrent = {};
+}
+
+window.qBittorrent.PiecesBar = (function() {
+    const exports = function() {
+        return {
+            PiecesBar: PiecesBar
+        };
+    };
+
+    const STATUS_DOWNLOADING = 1;
+    const STATUS_DOWNLOADED = 2;
+
+    let PiecesBars = 0;
+    const PiecesBar = new Class({
+        initialize: function(pieces, parameters) {
+            const vals = {
+                'id': 'piecesbar_' + (PiecesBars++),
+                'width': 0,
+                'height': 0,
+                'downloadingColor': 'green',
+                'haveColor': 'blue',
+                'borderSize': 1,
+                'borderColor': '#999'
+            };
+
+            if (parameters && $type(parameters) == 'object')
+                $extend(vals, parameters);
+            if (vals.height < 12)
+                vals.height = 12;
+
+            const obj = new Element('div', {
+                'id': vals.id,
+                'class': 'piecesbar_wrapper',
+                'styles': {
+                    'border': vals.borderSize + 'px solid ' + vals.borderColor,
+                    'height': vals.height,
+                }
+            });
+            obj.vals = vals;
+            obj.vals.pieces = $pick(pieces, []);
+
+            obj.vals.canvas = new Element('canvas', {
+                'id': vals.id + '_canvas',
+                'class': 'piecesbar_canvas',
+                'width': vals.width - (2 * vals.borderSize),
+                'height': 1
+            });
+            obj.appendChild(obj.vals.canvas);
+
+            obj.setPieces = PiecesBar_setPieces;
+            obj.refresh = PiecesBar_refresh;
+            obj.clear = PiecesBar_setPieces.bind(obj, []);
+            obj._drawStatus = PiecesBar_drawStatus;
+
+            if (vals.width)
+                obj.setPieces(vals.pieces);
+            else
+                setTimeout(function() { PiecesBar_checkForParent(obj.id); }, 1);
+
+            return obj;
+        }
+    });
+
+    function PiecesBar_setPieces(pieces) {
```

Comment:
Function naming should use camel case whenever possible.
It applies to other functions too.